### PR TITLE
code refactor: fstrings and paths

### PR DESCRIPTION
**Pull request by vinkenoogm**

### CODE REVIEW

File: src/7_impactbloodsupply.ipynb [hunk 1]
```diff
@@ -52,7 +70,7 @@
     "    return(intvals[i])\n",
     "\n",
     "def get_first_pos_pred(x, fulldf):\n",
-    "    firstcol = list(fulldf.columns).index('HbOK_pred_' + str(int(x['first_possible_invite'])))\n",
+    "    firstcol = list(fulldf.columns).index(f\"HbOK_pred_{int(x['first_possible_invite'])}\")\n",
```
Comment (sjvrijn) [inline src/7_impactbloodsupply.ipynb:73]: Is the `int()` call here still needed?

File: src/7_impactbloodsupply.ipynb [hunk 2]
```diff
@@ -62,8 +80,8 @@
     "def gather_results(sex, nbacks, foldersuffix=''):\n",
     "    df = list()\n",
     "    for nback in nbacks:\n",
-    "        scaler = pickle.load(open('../results/scalers'+foldersuffix+'/'+sex+'_'+str(nback)+'.pkl', 'rb'))\n",
-    "        dfsub = pd.read_pickle('../results/pred_timechange'+foldersuffix+'/data_res_'+sex+'_'+str(nback)+'.pkl')\n",
+    "        scaler = pickle.load(open(f'../results/scalers{foldersuffix}/{sex}_{nback}.pkl', 'rb'))\n",
+    "        dfsub = pd.read_pickle(f'../results/pred_timechange{foldersuffix}/data_res_{sex}_{nback}.pkl')\n",
```
Comment (sjvrijn) [inline src/7_impactbloodsupply.ipynb]: I think you can still replace the `../results` with `results_path / f'...'` from these paths

File: src/7_impactbloodsupply.ipynb [hunk 3]
```diff
@@ -126,19 +145,19 @@
...
-    "    print('  ', summ_int.loc[(0, '(90, 400]'), 'count'], '=', int(round(summ_int.loc[(0, '(90, 400]'), 'count']/tot_deferred*100, 0)), '% donors more than three months later')\n",
-    "    print('  ', summ_int.loc[(0, 'nan'), 'count'], '=', int(round(summ_int.loc[(0, 'nan'), 'count']/tot_deferred*100, 0)), '% donors not invited within year')\n",
+    "    print(f\"  {summ_int.loc[(0, '(-400, -14]'), 'count']} = {int(summ_int.loc[(0, '(-400, -14]'), 'count']/tot_deferred*100)}% donors more than two weeks earlier\")\n",
+    "    print(f\"  {summ_int.loc[(0, '(-14, 14]'), 'count']} = {int(summ_int.loc[(0, '(-14, 14]'), 'count']/tot_deferred*100)}% donors within two weeks\")\n",
+    "    print(f\"  {summ_int.loc[(0, '(14, 90]'), 'count']} = {int(summ_int.loc[(0, '(14, 90]'), 'count']/tot_deferred*100)}% donors two weeks - three months later\")\n",
+    "    print(f\"  {summ_int.loc[(0, '(90, 400]'), 'count']} = {int(summ_int.loc[(0, '(90, 400]'), 'count']/tot_deferred*100)}% donors more than three months later\")\n",
+    "    print(f\"  {summ_int.loc[(0, 'nan'), 'count']} =' {int(summ_int.loc[(0, 'nan'), 'count']/tot_deferred*100)}% donors not invited within year\")\n",
```
Comment (sjvrijn) [inline src/7_impactbloodsupply.ipynb]: I'd remove the dangling single quote from `...} =' {...` in this line

File: src/7_impactbloodsupply.ipynb [hunk 4]
```diff
@@ -229,17 +239,17 @@
...
-      "   515 = 13 % donors more than two weeks earlier\n",
-      "   373 = 9 % donors within two weeks\n",
-      "   422 = 11 % donors two weeks - three months later\n",
-      "   1584 = 40 % donors more than three months later\n",
-      "   1111 = 28 % donors not invited within year\n",
+      "  515 = 12% donors more than two weeks earlier\n",
+      "  373 = 9% donors within two weeks\n",
+      "  422 = 10% donors two weeks - three months later\n",
+      "  1584 = 39% donors more than three months later\n",
+      "  1111 =' 27% donors not invited within year\n",
```
Comment (sjvrijn) [inline src/7_impactbloodsupply.ipynb]: The dangling single quote from line 160 is showing up in the output here

**Review by sjvrijn — COMMENTED** [review]
Looks good! Just a few minor changes you may still want to make

File: src/7_impactbloodsupply.ipynb (hunk 1 above)
Comment (vinkenoogm) [reply src/7_impactbloodsupply.ipynb:73]: Still needed, as x['first_possible_invite'] is a float for some reason

File: src/7_impactbloodsupply.ipynb (hunk 2 above)
Comment (vinkenoogm) [reply src/7_impactbloodsupply.ipynb]: Replaced paths

**Pull request merged**